### DISCUSSION TITLE
how can I control where to put an equation with snippets?

### DISCUSSION THREAD

**Discussion by TheTacoSexyZ** [artisticat1/obsidian-latex-suite#259] [Q&A] [answered]
I apologize if this is a bad question, I'm new to this... :/

Here are two snippets examples
{trigger: "!", replacement: "\\dfrac{1}{${VISUAL}}", options: "mA"},
{trigger: "#", replacement: "x^{${VISUAL}", options: "mA"},

If I have this equation, x^2+\dfrac{1}{x}, I have to select it and then press # or ! (or any other characters) to trigger the action.

But what I am looking for, is that I can do the same, but without selecting anything, I just write any equation, and when I finish writing in that line and put # or !, then it activates any of the 2 actions 'my ecuation'**_!_**  to \dfrac{1}{my ecuation} or 'my ecuation'**_#_**  to x^{my ecuation}.. without having selected anything

**Comment by bdm-k** [comment, marked as answer]
Probably, the [regex snippet](https://github.com/artisticat1/obsidian-latex-suite/blob/main/DOCS.md#regex-snippets) can do what you want to do.
```
{ trigger: "([a-zA-Z])hat", replacement: "\\hat{[[0]]}", options: "rmA" }
```
For example, the above snippet will expand `xhat` to `\hat{x}`.
